src/engine/NodeHelpers.cs

```diff
@@ -69,6 +69,24 @@ public static void QueueFreeChildren(this Node node, bool detach = true)
         }
     }
 
+    /// <summary>
+    ///   Call RemoveChild on all Node children
+    /// </summary>
+    /// <param name="node">Node to remove children of</param>
+    public static void RemoveChildren(this Node node)
```

Comment:
What is this used for? We haven't needed such a method before, mainly because if you just remove the nodes without freeing them, you are leaking resources.

Reply:
I use it for generating the config menu as I am storing the node in a variable and then just adding them back to the tree when I need them

Reply:
Why don't you add a single parent node that contains that node tree, then you can easily attach / detach it?

Reply:
I didn't think of that and It was easier to integrate into my existing config creation system if i just saved the node in the config item

Reply:
If this method is kept, a massive warning needs to be slapped on this as this is not safe to use in the general case (very easily causes a resource leak). So I'd much prefer the single node attach / detach approach and removing this method.

Reply:
sure i do that

i remove the method and change it to just save the whole tree instead